### DISCUSSION TITLE
请问在使用umimax时，为什么使用了装饰器的代码无法被esbuild编译

### DISCUSSION THREAD

**Discussion by mazhenjie233** [umijs/umi#12425] [Q&A]
![1717002768276](https://github.com/umijs/umi/assets/76252062/d5366e91-37ed-467c-94b4-d922f74a5f2b)

**Comment by mazhenjie233** [comment]
<img width="572" alt="image" src="https://github.com/umijs/umi/assets/76252062/0bd733f9-4240-4ebe-a60b-c455ca67eac0">
<img width="532" alt="image" src="https://github.com/umijs/umi/assets/76252062/d865e4b6-d6a8-4646-a88a-7c072314f982">
插件与ts配置都加上了，强制升级esbuild为最新版本后转为<img width="865" alt="image" src="https://github.com/umijs/umi/assets/76252062/77c34f9b-ad0f-44cb-bc3f-2f53f8cfd794">babel与babel-loader报错

**Reply by xiaogonggong-w** [reply]
请问你解决这个问题了吗

**Reply by mazhenjie233** [reply]
后来没解决这个，只能不用这个语法，但我记得后面umi的升级记录里有支持accessor，现在没解决嘛

**Reply by xiaogonggong-w** [reply]
还是没有支持，我已经解决了，放弃这个语法

**Reply by xiaogonggong-w** [reply]
你这个编辑器用的是blocksuite吧

**Reply by mazhenjie233** [reply]
对

**Comment by fz6m** [comment]
给个最小复现吧，看你的报错是说语法有错误，不是装饰器的问题吧。

**Reply by mazhenjie233** [reply]
https://github.com/mazhenjie233/test-umi ,因为代码是复制的另一个开源代码的一部分，如果编译成功也会有一些其他引入报错，解决编译的问题就可以，感谢

**Reply by mazhenjie233** [reply]
`Support for the experimental syntax 'decoratorAutoAccessors' isn't currently enabled (23:20):

  21 |   `;
  22 |   @state()
> 23 |   private accessor show = true;
     |                    ^
  24 |
  25 |   @property({ attribute: false })
  26 |   accessor editor!: AffineEditorContainer;
    at constructor (C:\Users\mazhenjie\Desktop\poc\front-end\node_modules\@umijs\bundler-utils\compiled\babel\index.js:48920:19)`
esbuild升级到最新版后转为umi中的babel报错

**Comment by fz6m** [comment]
在 #12426 跟进，目测短时间内无法解决，请暂时避免这种 accessor 用法吧，最新版 umi 只用装饰器是可以的。
